### PR TITLE
Rework with skeleton DQN

### PR DESCRIPTION
**Pull request by Stroper111**
Based on your Stackoverflow question ([here](https://stackoverflow.com/questions/65262795/motion-planning-of-robots-using-reinforcement-learning)). I updated your code to be slightly more readable. 

Change log:
- There is no longer a Tkinter program, but I use [opencv-python](https://pypi.org/project/opencv-python/) for the rendering of the image. 
- There is now a 'RobotWorld' class that controls the world state and the rendering.
- The 'RobotEnv' class handles the interaction between the gym API and the 'RobotWorld'.
- The map creation has been made simpler, and an example can be viewed in the 'helper.constants' module.
- There is a skeleton code file in the folder 'agent', that contains the step that you have to do, in order to create a DQN agent.
- There are now 2 observation modes, grid and image, the grid is a smaller observation (9x9) against the full image(360, 360, 3) and is a better point to start with (called by 'robot-v1').
- Updated the 'setup.py' file, with 'opencv-python', my name and a new version.
- Added an example folder with basic gym loop.

### CODE REVIEW

File: gym_robot/envs/robot_env.py [hunk 1]
```diff
@@ -1,50 +1,72 @@
...
-        next_state = self.environment.step(action)
-        reward = self.environment.step(action)
-        done = self.environment.step(action)
-        return  next_state, reward, done, {}
-    
-    def render(self, mode="human", close=False):
-        self.environment.render()
-    
+        self.world.reset()
+        return self.world.image()
```
Comment (Stroper111) [inline gym_robot/envs/robot_env.py]:
```suggestion
        return self._get_obs(self.obs_mode)
```

File: gym_robot/envs/robot_env.py [hunk 2]
```diff
@@ -1,50 +1,72 @@
...
+
+    def render(self, mode="human", delay=1):
+        if mode == 'human':
+            return self.world.render(delay=delay)
+        if mode == 'rgb_array':
+            return self.world.image()
```
Comment (Stroper111) [inline gym_robot/envs/robot_env.py]:
```suggestion
            return self._get_obs(self.obs_mode)
```

File: gym_robot/envs/robot_world.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+class RobotWorld:
+    """
+        Represent the robot world environment.
+
+        :param Tuple[int, int] size:
+            This is the grid size of the world. Where the first
+            value is the number of grids on the x axis, and the second
+            for the y axis.
+
+        :param Union[None, Dict[Tuple[int, int], str]} mapping:
```
Comment (Stroper111) [inline gym_robot/envs/robot_world.py]:
```suggestion
        :param Union[None, Dict[Tuple[int, int], str]] mapping:
```

**Review by Stroper111 — COMMENTED** [review]
Looks fine to me

**Review by zoraiz-ali — APPROVED** [review]

**Comment by zoraiz-ali** [comment]
Hello Stroper111 please check this.
when i run this code

env = gym.make('robot-v1')
state_size = env.observation_space.shape[0]
action_size = env.action_space.n

ModuleNotFoundError come,

ModuleNotFoundError: No module named 'gym_robot.envs.helper'

**Comment by zoraiz-ali** [comment]
> Looks fine to me

Kindly check, I got the error module not found.

**Comment by zoraiz-ali** [comment]
Stroper111, I look forward to your reply

**Comment by Stroper111** [comment]
I checked, for me there is no error. Did you mark your project as the source directory? Because it seems that the top level imports are going wrong.

**Pull request merged**